subsys/net/lib/nrf_cloud/Kconfig.nrf_cloud_agps

```diff
@@ -3,8 +3,39 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
-config NRF_CLOUD_AGPS
+menuconfig NRF_CLOUD_AGPS
 	bool "nRF Cloud Assisted GPS (A-GPS)"
 	depends on MODEM_INFO
 	depends on MODEM_INFO_ADD_NETWORK
 	select CJSON_LIB
+
+if NRF_CLOUD_AGPS
+
+config NRF_CLOUD_AGPS_DISABLE_SCHEDULED_DOWNLOADS
```

Comment:
Just below in this file, I select it on when NRF_CLOUD_AGPS_FILTERED is enabled, because I was under the impression that when filtered ephemerides are used, this scheduled downloads *should always* be disabled.  In your review comments today you are saying there are caveats to that, which is news to me.

Reply:
When I said that scheduled downloads should always be disabled with filtered ephemerides, I was thinking about the periodic mode, because periodic is the only mode where GNSS does scheduled downloads. In continuous tracking mode GNSS is running and receiving all the time, so there's no need for additional download logic. And in single fix mode GNSS stops after producing a single fix and doesn't do anything after that until it's started again. So, scheduled downloads only apply to the periodic mode and setting the corresponding use case bit makes no difference in other modes (it doesn't hurt either).

My point about the NRF_CLOUD_AGPS_DISABLE_SCHEDULED_DOWNLOADS option is that as an application developer I would expect scheduled downloads to be disabled if I set the option in prj.conf, but actually setting the configuration option does nothing. It exists only so that the application can check if it's enabled and act accordingly. I think it would be clearer if the application made the decision based on something else, for example based on CONFIG_NRF_CLOUD_AGPS or CONFIG_NRF_CLOUD_AGPS_FILTERED. The documentation for filtered ephemerides should say that GNSS scheduled downloads must be disabled (if periodic mode is used).